File: src/main/java/com/example/domain/festival/event/FestivalSyncEventPublisher.java
package com.example.domain.festival.event;

import lombok.RequiredArgsConstructor;
import org.springframework.context.ApplicationEventPublisher;
import org.springframework.stereotype.Component;

import java.util.List;


//축제 목록 동기화 완료 이벤트 발행 클래스
@Component
@RequiredArgsConstructor
public class FestivalSyncEventPublisher {

    private final ApplicationEventPublisher applicationEventPublisher;

    //목록 동기화 완료 후, 변경된 contentId 목록을 이벤트로 발행한다.
    public void publishSyncCompleted(List<String> changedContentIds) {
        applicationEventPublisher.publishEvent(new FestivalSyncCompletedEvent(changedContentIds));
    }
}

File: src/main/kotlin/com/example/domain/festival/event/FestivalSyncEventPublisher.kt
package com.example.domain.festival.event

import com.example.domain.festival.dto.response.FestivalSyncResultResponse
import org.springframework.context.ApplicationEventPublisher
import org.springframework.stereotype.Component

// 축제 목록 동기화 완료 이벤트 발행 클래스
@Component
class FestivalSyncEventPublisher(
    private val applicationEventPublisher: ApplicationEventPublisher
) {
    // 목록 동기화 완료 후, 변경된 contentId 목록을 이벤트로 발행한다.
    fun publishSyncCompleted(
        changedContentIds: List<String>,
        listResult: FestivalSyncResultResponse
    ) {
        applicationEventPublisher.publishEvent(
            FestivalSyncCompletedEvent(changedContentIds, listResult)
        )
    }
}